### ISSUE TITLE
Could a DataFrameRow be shown as a record rather than as a table?

### ISSUE THREAD

**Issue by akosuas** [JuliaData/DataFrames.jl#2616]
Currently a single row is still printed as a "DataFrame" - i.e. a 1-row table, with truncated columns. I wonder if it would be more natural to show a single row (a "record") vertically, with one row per field? This would make it much easier to inspect a single record that has a lot of fields/columns.

**Comment by bkamins** [comment]
We could, but it would probably require a completely separate code path for this.

@ronisbr - any opinion on this?

**Comment by tcovert** [comment]
Note, this would be similar to `dplyr`'s `glimpse` function which I quite like...

**Comment by pdeffebach** [comment]
This also seems nice. 

Tables.jl has a row type which prints like a named tuple. We could change upstream and copy their printing as well.

**Comment by quinnj** [comment]
Yes, if anyone wants to improve `AbstractRow`/`AbstractColumns` default printing, I would love a PR: https://github.com/JuliaData/Tables.jl/blob/bdde6d343a9717cf78c7b442d142a9699688dbdc/src/Tables.jl#L193

**Comment by bjarthur** [comment]
worth noting that this is what pandas does:

```
>>> row
Unnamed: 0                                AAACCTGAGGGCTCTC-300.1.1
cell                                      AAACCTGAGGGCTCTC-300.1.1
n.umi                                                         2319
time.point                                             300_minutes
batch                                        Waterston_300_minutes
Size_Factor                                                1.45821
cell.type                                                      NaN
cell.subtype                                                   NaN
plot.cell.type                                                 NaN
raw.embryo.time                                                260
embryo.time                                                  220.0
embryo.time.bin                                            210-270
raw.embryo.time.bin                                        210-270
lineage                                                   MSxapaap
passed_initial_QC_or_later_whitelisted                        True
Name: 1, dtype: object
```

**Comment by bkamins** [comment]
Thank you - as commented in Tables.jl I think that the key decision is if we print data types. I think it is OK to skip them (as e.g. column can have type `Any`, while a specific value will have some concrete type and it could be confusing what was printed).

**Comment by bkamins** [comment]
@ronisbr - you could also take a look at the list https://github.com/JuliaData/DataFrames.jl/issues?q=is%3Aissue%20state%3Aopen%20label%3Adisplay and decide which of the things it would make sense to resolve in the process of porting to PrettyTables.jl v3. Thank you!

**Comment by bjarthur** [comment]
the temporary hack until single rows of a dataframe are printed vertically, is to use TerminalPager.jl.  `df |> pager` and then the arrow keys to scroll left and right.

**Comment by bjarthur** [comment]
one can also `[names(df) [values(df)...]]`

**Comment by ronisbr** [comment]
Hi!

Since v3 of PrettyTables.jl is now integrated, we just need to check what we need to do in those cases.

**Comment by bjarthur** [comment]
have we decided to print data types?  since they are printed for the current table output, i'd suggest they should alse be printed when transposing a single row into a record.

my motivation here is just to make it easier to see all the columns when there are many.  better use of screen real estate.

**Comment by pdeffebach** [comment]
I just realized I made that PR. I'm happy to add the types back.

**Comment by bkamins** [comment]
If I remember correctly the issue was whether to show column type or actual data type in a given cell.

**Comment by bjarthur** [comment]
the actual data type should be obvious or at least inferable by who ever is looking at the terminal.  but column type wouldn't be.  so i'd suggest the latter.

more generally, it'd be nice to have a way to similarly "transpose" a table with more than one row.  for example, say i take a subset of a dataframe and the result has only two rows.  were it to have a lot of columns, it might fit better on screen with columns listed vertically.  maybe in another PR/issue we could discuss adding a `transpose` kwarg to `show`.
